### PR TITLE
Fixed DHCP Hostnames and Better Bandwidth Controls

### PR DESCRIPTION
**Pull request by TNT-Samuel**
The rpc call returns uppercase mac addresses, I have now made it return lowercase mac addresses which also allows uppercase mac addresses in the user file

I have also changed the bandwidth control into a config with seperate upstream/downstream values

I have also added an option to view speeds in bits/second instead of bytes for people who prefer to see it that way

### CODE REVIEW

**Comment by brvphoenix** [comment]
@TNT-Samuel 
I have no enough time to review all the changes now. Maybe, I'll give you a feedback before next week.

**Comment by TNT-Samuel** [comment]
How is the code review going?

**Comment by brvphoenix** [comment]
@TNT-Samuel 
1. The changes for fixing the dhcp hostname resolutions hould be accepted immediately. I'll cherry-pick this commit if you don't mind this, or you can split it in a separate commit.
1. It seems that the DSL bandwidth is not a universal option for all use cases. I'm not using DSL, so I can't test this. Besides, these codes may need to be optimized.
1. I have no opinion on the option of displaying bandwidth in bits.
1. Personally, I think saving the settings in `/etc/config/wrtbwmon` is not a good choice, because the  files in `/etc/config/*` is mainly used to configure the daemon and it will trigger the daemon reload once it is changed. I suggest that these settings can be saved in `/etc/luci-wrtbwmon.conf` or in the browser cache with `window.localstorage`.
1. The different usages between KB (1000) and KiB (1024) are confusing. Considering the units for bandwidth, it is reasonable to use KB rather than KiB. [But the OpenWrt seems to use KiB](https://github.com/openwrt/luci/blob/219f0ce891a2e59d28a2dd8a1d14a29e35a81185/modules/luci-mod-status/htdocs/luci-static/resources/view/status/bandwidth.js#L24).  I'm not insist on this.

**Comment by brvphoenix** [comment]
> Do you want me to add a config option for choosing betweeen KB and KiB?

LGTM

> Any feedback would be great!

There seems to be a lot of duplicate code. I will merge this pull request after simplifying these codes. You can wait for me or implement it yourself.

**Comment by TNT-Samuel** [comment]
New changes done!
Anything else?

**Comment by TNT-Samuel** [comment]
I have tested it and it seems to work great!

**Comment by TNT-Samuel** [comment]
What happened?
Do you want me to open a new pull request?

**Comment by brvphoenix** [comment]
No, I have squashed the commit. But it seems that I have made someting wrong.
I'm so sorry.
Your pull request is saved in my local repo and everything is OK.

**Comment by TNT-Samuel** [comment]
Can I submit a new request as I'm doing a competition which one of the tasks is to contribute to a public repository?

**Comment by TNT-Samuel** [comment]
Oh this works instead!

**Comment by brvphoenix** [comment]
> Can I submit a new request as I'm doing a competition which one of the tasks is to contribute to a public repository?

No.

**Comment by TNT-Samuel** [comment]
> > Can I submit a new request as I'm doing a competition which one of the tasks is to contribute to a public repository?
> 
> No.

Sorry, I didn't know you could reopen pull requests :)

**Comment by brvphoenix** [comment]
You can check my modified commits. I just squash some commits with the same goal.

File: luci-app-wrtbwmon/htdocs/luci-static/resources/view/wrtbwmon/details.js [hunk 1]
```diff
@@ -76,28 +150,20 @@ function parseDatabase(values, hosts) {
...
-	    fv = byte ? String.format('%1024.2mB', v) : v,
+	    vn = v || 0,
+	    mn = m || 100,
+	    fv = formatSpeedNative(v),
 	    pc = ((100 / mn) * vn).toFixed(2),
 	    wt = Math.floor(pc > 100 ? 100 : pc),
-	    bgc = (pc >= 95 ? 'red' : (pc >= 80 ? 'magenta' : (pc >= 60 ? 'yellow' : 'lime')));
-
+	    bgc = (pc >= 95 ? 'red' : (pc >= 80 ? 'darkorange' : (pc >= 60 ? 'yellow' : 'lime')));
+		var tc = (pc >= 80 ? 'white' : '#404040');
```
Comment (brvphoenix) [inline luci-app-wrtbwmon/htdocs/luci-static/resources/view/wrtbwmon/details.js:161]: The indentation seems incorrect.
You can write as
```suggestion
	    bgc = (pc >= 95 ? 'red' : (pc >= 80 ? 'darkorange' : (pc >= 60 ? 'yellow' : 'lime'))),
	    tc = (pc >= 80 ? 'white' : '#404040');
```

File: luci-app-wrtbwmon/htdocs/luci-static/resources/view/wrtbwmon/details.js [hunk 2]
```diff
@@ -25,6 +37,68 @@ function getPath() {
 	});
 }
 
+function getDisplayConfigValue(config_name, default_value) {
```
Comment (brvphoenix) [inline luci-app-wrtbwmon/htdocs/luci-static/resources/view/wrtbwmon/details.js:40]:
```suggestion
function getDisplayConfigValue(option, default) {
```

File: luci-app-wrtbwmon/htdocs/luci-static/resources/view/wrtbwmon/details.js [hunk 3]
```diff
@@ -543,8 +642,19 @@ return L.view.extend({
...
+				bitsorBytes(),
+				useDSLBandwidth(),
+				downstreamBandwidth(),
+				upstreamBandwidth()]).then(function() {
+					updateBandwidths();
+					L.Poll.add(updateData, $('intervalSelect').value);
+					L.Poll.add(updatePerSec, 1);
+		
+					registerTableEventHandlers();
+				});
```
Comment (brvphoenix) [inline luci-app-wrtbwmon/htdocs/luci-static/resources/view/wrtbwmon/details.js:657]:
```suggestion
		Promise.all([
				use1024Bytes()
				bitsorBytes(),
				useDSLBandwidth(),
				downstreamBandwidth(),
				upstreamBandwidth()])
		]).then(function() {
			updateBandwidths();
			L.Poll.add(updateData, $('intervalSelect').value);
			L.Poll.add(updatePerSec, 1);
		
			registerTableEventHandlers();
		});
```

**Comment by brvphoenix** [comment]
In fact, you can use
`git add .`
`git  commit --amend`
`git push -f`
to modify the commit you have pushed.

**Comment by brvphoenix** [comment]
@TNT-Samuel 
Don't modify your repo for now. I'll resolve the conflicts.

File: luci-app-wrtbwmon/htdocs/luci-static/resources/view/wrtbwmon/details.js (hunk 3 above)
Comment (brvphoenix) [reply luci-app-wrtbwmon/htdocs/luci-static/resources/view/wrtbwmon/details.js:657]: @TNT-Samuel
I fixed some syntax errors. You can test this.

**Comment by TNT-Samuel** [comment]
I have merged your changes with a few modifications to get them to work
 - "default" is a reserved keyword in js
 - The upstream/downstream config load functions need to be run after use1024Bytes as it sets the multiplier 1000/1024 that they use

**Comment by brvphoenix** [comment]
This pull request will be merged after I refactor the functions to save default settings. The current screenshot is as follows which will storage the conf file in other path rather than '/etc/config'.
捕获

**Comment by TNT-Samuel** [comment]
Thanks for the feedback and help!

**Pull request merged**